pulsar-broker/src/test/java/org/apache/pulsar/broker/stats/PrometheusMetricsTest.java

```diff
@@ -1634,6 +1643,29 @@ public static Multimap<String, Metric> parseMetrics(String metrics) {
         return parsed;
     }
 
+    @Test
+    public void testRawMetricsProvider() throws IOException {
+        PrometheusMetricsProvider rawMetricsProvider = new PrometheusMetricsProvider();
+        rawMetricsProvider.start(new PropertiesConfiguration());
+        rawMetricsProvider.getStatsLogger("test").getOpStatsLogger("test_metrics")
+            .registerSuccessfulEvent(100, TimeUnit.NANOSECONDS);
+
+        getPulsar().addPrometheusRawMetricsProvider(rawMetricsProvider);
```

Comment:
If you already have integration built it, and you go about querying the real metrics endpoint, why not check that you have a real BK client metric *with* the labels you expect to have but now you actually have them?

Reply:
The PR is not mainly developed for the BK client metrics, and it is used for the plugin metrics integrated with Pulsar broker.

Reply:
I don't have enough knowledge on 
> is used for the plugin metrics integrated with Pulsar broker.

Can you please elaborate and give references?

Reply:
Oh, I read your code below. 
I'm still amazed after spending weeks reading all the metrics code, I'm still discovering new bits.
I was wondering what was that weird `PrometheusRawMetricsProvider` nobody was actually using inside Pulsar.
So you're saying it's in fact a public API for other plugin developers to rely on that? Can you please give me a reference for such one?

Reply:
Yes, it is a public API for other plugins which run within the Pulsar broker to expose metrics to the Pulsar broker metric port. A public plugin repo is KOP. https://github.com/streamnative/kop/blob/a9f56e9b0435429dd8d977ba948c6772e6fe5b86/kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaProtocolHandler.java#L280. 
I'm developing another plugin, which also needs this public API to expose metrics.

Reply:
@hangc0276 Ok, from my understanding, and correct me if I'm wrong here: This PR fixes the BK Metrics API implementation in such a way that it will support labels. *But* it doesn't fix another issue here: The ability to *use* `PrometheusMetricsProvider` by plugins. 

So today, and in your PR, we only print the metrics registered to `PrometheusMetricsProvider` instantiated in `ManagedLedgerFactory` - which is good as we now get the BK client metrics with labels.

*But*, I believe the correct solution should be to instantiate `PrometheusMetricsProvider` in `PulsarService`, and pass it along to `ManagedLedgerFactory` and also available to retrieve by any plugin if needed.

The way I see it, plugins should use:
1. Prometheus Client library - preferred.
2. BK Metrics API implementation - not the best-preferred way. 
3. Raw metrics provider, if all other means fail.
If we're already doing this, at least let's do this right. In your case, you know you want to use the `PrometheusMetricsProvider` maybe we should just expose it and use it properly.